tests/sqllogictests/suites/mode/standalone/explain_native/infer_filter.test

```diff
@@ -541,6 +541,94 @@ HashJoin
         ├── push downs: [filters: [and_filters(t1.a (#0) > 5, t1.a (#0) < 10)], limit: NONE]
         └── estimated rows: 0.00
 
+# t1.a > 5 and t2.a > 10
+query T
+explain select * from t1, t2 where t1.a > t2.a and t1.a > 5 and t2.a > 10;
+----
+MergeJoin
+├── output columns: [t2.a (#2), t2.b (#3), t1.a (#0), t1.b (#1)]
+├── join type: INNER
+├── range join conditions: [t2.a (#2) "lt" t1.a (#0)]
+├── other conditions: []
+├── estimated rows: 0.00
+├── TableScan(Left)
+│   ├── table: default.default.t2
+│   ├── output columns: [a (#2), b (#3)]
+│   ├── read rows: 0
+│   ├── read bytes: 0
+│   ├── partitions total: 0
+│   ├── partitions scanned: 0
+│   ├── push downs: [filters: [t2.a (#2) > 10], limit: NONE]
+│   └── estimated rows: 0.00
+└── TableScan(Right)
+    ├── table: default.default.t1
+    ├── output columns: [a (#0), b (#1)]
+    ├── read rows: 0
+    ├── read bytes: 0
+    ├── partitions total: 0
+    ├── partitions scanned: 0
+    ├── push downs: [filters: [t1.a (#0) > 5], limit: NONE]
+    └── estimated rows: 0.00
+
+# t1.a > 10
+query T
+explain select * from t1, t2 where t1.a > t2.a and t1.a > 5 and t1.a > 10;
+----
+MergeJoin
+├── output columns: [t2.a (#2), t2.b (#3), t1.a (#0), t1.b (#1)]
+├── join type: INNER
+├── range join conditions: [t2.a (#2) "lt" t1.a (#0)]
+├── other conditions: []
+├── estimated rows: 0.00
+├── TableScan(Left)
+│   ├── table: default.default.t2
+│   ├── output columns: [a (#2), b (#3)]
+│   ├── read rows: 0
+│   ├── read bytes: 0
+│   ├── partitions total: 0
+│   ├── partitions scanned: 0
+│   ├── push downs: [filters: [], limit: NONE]
+│   └── estimated rows: 0.00
+└── TableScan(Right)
+    ├── table: default.default.t1
+    ├── output columns: [a (#0), b (#1)]
+    ├── read rows: 0
+    ├── read bytes: 0
+    ├── partitions total: 0
+    ├── partitions scanned: 0
+    ├── push downs: [filters: [t1.a (#0) > 10], limit: NONE]
+    └── estimated rows: 0.00
+
+# false
+query T
+explain select * from t1, t2 where t1.a = t2.a and t1.a > 5 and t2.a < 1;
```

Comment:
Filter false is pushed down probe side and still need to build hash table, in the future, maybe we can push down filter both build and probe side?

Reply:
Yes, when the condition is false, we need a dummy operator to replace the original one, but some problems encountered and it will be added in the next PR.